types/sap__approuter/index.d.ts

```diff
@@ -0,0 +1,428 @@
+         *               incomingResponse.setHeader('header1', 'abc');
+         *               done(null, incomingResponse);
+         *           } else {
+         *               done('An error occurred in backend, returned status ' + ctx.outgoingResponse.statusCode, ctx.incomingResponse);
+         *           }
+         *       };
+         *       next();
+         *   });
+         *   ar.start();
+         * @param ctx - Context object for the request handler
+         * @param {AppRouterIncomingMessage~afterRequestHandlerCallback} done - A callback function that receives (optionally) and error and the modified incomingResponse
+         */
+        afterRequestHandler?: (
+            ctx: AfterRequestHandlerContext,
+            done: (error: Error | string | undefined | null, incomingResponse: ServerResponse) => void,
+        ) => void;
+
+        /**
+         * A function that can be added to the request object - for example in a "first" or "beforeRequestHandler" extension.
+         * If exists, this function will be called by the standard application router when a backend connection timeout occurs.
+         *
+         * @param req - the request object
+         * @param done - a callback function that doesn't return any parameter
+         */
+        backendTimeout?: (req: IncomingMessage, done: () => void) => void;
+
+        /**
+         * Callback for {@link afterRequestHandler}
+         *
+         * @param error - Error object in case of error
+         * @param incomingResponse - Modified incomingResponse
+         */
+    }
+
+    /**
+     * A handler for requests, called Middleware.
+     *
+     * @param request - Data of the incoming Request
+     * @param response - Object for the outgoing Response; can be manipulated to alter the response
+     * @param next - Call next to give control back to Application Router Middleware
+     */
+    type MiddlewareHandler = (request: AppRouterIncomingMessage, response: ServerResponse, next: () => void) => void;
+
+    interface Extensions {
+        /**
+         * Describes the middleware provided by this extension
+         * If a path is given, the handler is only invoked on these paths; otherwise it's invoked on all requests.
+         */
+        insertMiddleware: {
+            /** A MiddlewareSlot before the first application router middleware */
+            first?: Array<MiddlewareHandler | { path: string; handler: MiddlewareHandler }>;
+            /** A MiddlewareSlot before the standard application router request handling */
+            beforeRequestHandler?: Array<MiddlewareHandler | { path: string; handler: MiddlewareHandler }>;
+            /** A MiddlewareSlot before the standard application router error handling */
+            beforeErrorHandler?: Array<MiddlewareHandler | { path: string; handler: MiddlewareHandler }>;
+        };
+    }
+
+    type RouterConfig = unknown;
+
+    class StartOptions {
```

Comment:
This class cannot be exposed through namespace because it doesnt exist at runtime:
```js
//a.js
const x = require("@sap/approuter");
console.log(x.StartOptions )
```
```
> node a.js
#2.0#2023 11 03 10:43:40:835#-07:00#WARNING#/LoggingLibrary################PLAIN##Dynamic log level switching not available#
undefined
```

Reply:
changed it to be an interface